src/osd/PG.cc

```diff
@@ -1168,7 +1168,8 @@ void PG::calc_ec_acting(
     if (acting.size() > (unsigned)i && acting[i] != CRUSH_ITEM_NONE &&
 	!all_info.find(pg_shard_t(acting[i], shard_id_t(i)))->second.is_incomplete() &&
 	all_info.find(pg_shard_t(acting[i], shard_id_t(i)))->second.last_update >=
-	auth_log_shard->second.log_tail) {
+	auth_log_shard->second.log_tail &&
+        std::find(want.begin(), want.end(), acting[i]) == want.end()) {
```

Comment:
Thanks for pointing out this for me, kefu. I confess that I don't understand the behaviour of EC-PGs well:-(